File: fastapi-python-backend/src/routes/metrics_route.py
from fastapi import APIRouter, Response

from ..config.env import settings
from ..errors.http_errors import NotFoundError
from ..observability.metrics import get_metrics_snapshot, metrics_content_type

metrics_router = APIRouter(prefix='/metrics', tags=['metrics'])


@metrics_router.get('', include_in_schema=True)
async def get_metrics() -> Response:
    if not settings.METRICS_ENABLED:
        raise NotFoundError(detail='Metrics endpoint is disabled.')

    metrics = get_metrics_snapshot()
    return Response(content=metrics, media_type=metrics_content_type)

File: express-ts-bun-backend/src/routes/metrics.route.ts
import { Router } from 'express';
import { env } from '../config/env';
import { NotFoundError } from '../errors/http-errors';
import { getMetricsSnapshot, metricsRegistry } from '../observability/metrics';

const metricsRouter = Router();

metricsRouter.get('/', async (_req, res, next) => {
  if (!env.METRICS_ENABLED) {
    next(
      new NotFoundError({
        detail: 'Metrics endpoint is disabled.',
      }),
    );
    return;
  }

  try {
    const metrics = await getMetricsSnapshot();
    res.setHeader('Content-Type', metricsRegistry.contentType);
    res.status(200).send(metrics);
  } catch (error) {
    next(error);
  }
});

export { metricsRouter };
